### PR TITLE
Support telnet command to view ark plugin

### PR DESCRIPTION
**Pull request by JoeKerouac**
Support telnet command to view ark plugin #79

### CODE REVIEW

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/PluginCommandProvider.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                    Plugin plugin = service.getPluginByName(commands[2]);
+                    if (plugin == null) {
+                        sb.append(String.format("plugin %s not found", commands[2]));
+                    } else {
+                        sb.append(getPrint(plugin));
+                    }
+                    break;
+                }
+            default:
+                sb.append("command error\n");
```
Comment (QilongZhang) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/PluginCommandProvider.java]: `command error` should be evaluated in validate phase.

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/AbstractCommandProvider.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Override
+    public boolean validate(String command) {
+        if (StringUtils.isEmpty(command)) {
+            return false;
+        }
+
+        String[] strs = process(command);
+        command = strs[0];
+        return command.equals(getCommandPre());
```
Comment (QilongZhang) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/AbstractCommandProvider.java]: the command validation is a litter sample,  also command parameter should  be validated.

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/AbstractCommandProvider.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    /**
+     * deal commandLine(split by blank character)
+     * @param commandLine commandLine
+     * @return Every valid character after processing
+     */
+    protected String[] process(String commandLine) {
+        return commandLine.trim().split(Constants.SPACE_SPLIT);
+    }
```
Comment (QilongZhang) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/AbstractCommandProvider.java]: it's unnecessary to create a new `process` method.

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/AbstractCommandProvider.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+        return handleCommandLine(commands);
+    }
+
+    @Override
+    public String getHelp(String commandMarker) {
+        if (getCommandPre().equals(commandMarker)) {
+            return getHelp();
+        } else {
+            return null;
```
Comment (QilongZhang) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/AbstractCommandProvider.java]: `null` should not be returned, `StringUtils.EMPTY_STRING` is better

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/PluginCommandProvider.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        StringBuilder sb = new StringBuilder();
+        sb.append("plugin list : list ark container all plugins name.\n");
+        sb.append("plugin info [plugin-name] : list plugin info.\n");
+        return sb.toString();
+    }
+
+    private String getPrint(Plugin plugin) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(plugin.getPluginName()).append(" :").append("\n");
+        sb.append("\t").append("group-id : ").append(plugin.getPluginName()).append("\n");
```
Comment (QilongZhang) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/PluginCommandProvider.java]: `\t` shows different in windows and mac, space is better

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/pipeline/RegisterServiceStage.java [hunk 6]
```diff
@@ -50,6 +52,8 @@ public void process(PipelineContext pipelineContext) throws ArkException {
     private void registryDefaultService() {
         registryService.publishService(BizDeployer.class, new DefaultBizDeployer(),
             new ContainerServiceProvider());
+        registryService.publishService(CommandProvider.class, new PluginCommandProvider(),
+            new ContainerServiceProvider());
```
Comment (QilongZhang) [inline sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/pipeline/RegisterServiceStage.java]: it's better use a uniqueId to publish a service, as maybe multi implementation of `CommandProvider` would be supported.

File: sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/PluginCommandProvider.java (hunk 1 above)
Comment (JoeKerouac) [reply sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/command/PluginCommandProvider.java]: This line will not run now, but if do not add this default there will be warning.

**Pull request closed without merging**